service/dap/server.go

```diff
@@ -1023,9 +1044,58 @@ func (s *Session) onLaunchRequest(request *dap.LaunchRequest) {
 	argsToLog.Cwd, _ = filepath.Abs(args.Cwd)
 	s.config.log.Debugf("launching binary '%s' with config: %s", debugbinary, prettyPrint(argsToLog))
 
+	var redirected = false
+	s.outputMode |= outputToStd
+	switch args.OutputMode {
+	case "remote":
+		s.outputMode = outputToDAP
+		redirected = true
+	case "local", "":
+		// noting
+	default:
+		s.sendShowUserErrorResponse(request.Request, FailedToLaunch, "Failed to launch",
+			fmt.Sprintf("invalid debug configuration - unsupported 'outputMode' attribute %q", args.OutputMode))
+		return
+	}
+
+	readerFunc := func(reader io.Reader, category string) {
+		var stdWriter io.Writer
+		if category == "stdout" {
+			stdWriter = os.Stdout
+		} else {
+			stdWriter = os.Stderr
+		}
+
+		var out [1024]byte
+		for {
+			n, err := reader.Read(out[:])
+			if err != nil {
+				if errors.Is(io.EOF, err) {
+					return
+				}
+				s.config.log.Errorf("failed read by %s - %v ", category, err)
+				return
+			}
+			outs := string(out[:n])
+
+			if s.outputMode&outputToStd != 0 {
```

Comment:
This can never happen now.

Reply:
yes.I will remove the part related to `outputMode` that has `redirected` enough.

Reply:
done.